### ISSUE TITLE
Crash on invalid input with -s flag

### ISSUE THREAD

**Issue by quantumech3** [Qalculate/libqalculate#324]
Hello!
When I run `qalc -s update_exchange_rates "det([["` or qalc -s `update_exchange_rates "det(["` I get the following error:    
```
[1]    12503 segmentation fault (core dumped)  qalc -s update_exchange_rates "det([["
```
I am running Ubuntu 20.04 and using the qalc apt package. I originally discovered this issue when using rofi-calc which builds off of this project and that issue is linked here: https://github.com/svenstaro/rofi-calc/issues/72

**Comment by hanna-kn** [comment]
The crash is caused by the empty matrix. This was fixed more than two years ago in version 3.2.

**Comment by quantumech3** [comment]
Oh ok, apologies haha.
